### ISSUE TITLE
🤗 [Question]: 

### ISSUE THREAD

**Issue by Krishnakumar-b** [gofiber/fiber#2466]
### Question Description

I was trying to get the content length for response. I am getting 0 as the value.But when I get the response in client , I am getting content length.

I was using the method : c.Response().Header.ContentLength() , and also tried with c.Context().Response.Header.ContentLength() , the result is 0.

Can I please get a response?

Thank you

### Code Snippet (optional)

```go
fmt.Println("Length",c.Response().Header.ContentLength())
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my questions prior to opening this one.
- [X] I understand that improperly formatted questions may be closed without explanation.

**Comment by leonklingele** [comment]
You can probably add a middleware to achieve this:

```go
app := fiber.New()

app.Use(func(ctx *fiber.Ctx) error {
    if err := c.Next(); err != nil {
        return err
    }

    fmt.Println("Length", c.Response().Header.ContentLength())
})
```

**Comment by Krishnakumar-b** [comment]
> Thanks for the reponse, But I am doing the same

func limit(c *fiber.Ctx) (err error) {
	if c.Response().Body() == nil && c.Request().Body() != nil {
		err = c.Next()
		if err != nil {
			return
		}
	}
	fmt.Println("Length",c.Response().Header.ContentLength())
	return c.Next()
}

**Comment by leonklingele** [comment]
Sorry, only now got to try this myself.

The `logger` middleware determines the length as follows: https://github.com/gofiber/fiber/blob/182f9f09705eab40c61a618835d46faee79c1e49/middleware/logger/tags.go#L90-L95

Also, this seems to work:

```go
package main

import (
	"log"
	"net"

	"github.com/gofiber/fiber/v2"
)

func main() {
	app := fiber.New()

	app.Use(func(c *fiber.Ctx) error {
		res := c.Response()
		c.Context().Hijack(func(_ net.Conn) {
			log.Printf(
				"Body=%s, length=%d\n",
				res.Body(),
				res.Header.ContentLength(),
			)
		})

		return c.Next()
	})

	app.Get("/", func(c *fiber.Ctx) error {
		return c.SendString("Hello, World 👋!")
	})

	if err := app.Listen("127.0.0.1:8080"); err != nil {
		log.Panicln(err)
	}
}
```

Although doing so you'll not be able to determine the number of compressed bytes sent if compression is used.

Does that help @Krishnakumar-b ?

**Comment by Krishnakumar-b** [comment]
Thanks for the response @leonklingele , I will have a check with this as well.

**Comment by ReneWerner87** [comment]
yes because fasthttp is so lazy that they do the calculation of the length only at the end
https://github.com/valyala/fasthttp/blob/9bc8e480c444fa6bf54a4cd79091692ceff6011c/http.go#L1447

```go
package main

import (
	"fmt"
	"github.com/gofiber/fiber/v2"
)

func main() {
	app := fiber.New()

	app.Use(func(c *fiber.Ctx) error {
		if err := c.Next(); err != nil {
			return err
		}

		fmt.Println("Length", len(c.Response().Body()))

		return nil
	})

	app.Get("/test", func(c *fiber.Ctx) error {
		return c.SendString("abc")
	})

	app.Listen(":3000")
}

```

**Comment by Krishnakumar-b** [comment]
This will work; Thanks @ReneWerner87
